include/Surelog/Common/PathId.h

```diff
@@ -22,6 +38,19 @@
 #endif
 
 namespace SURELOG {
+/**
+ * class PathId
+ *
+ * Used to uniquely represent a file or directory abstractly.
+ * The context/value doesn't have to be a std::filesystem::path but
+ * can be any printable (i.e. convertible to string) value. This pinned
+ * value is stored in the PathId::m_symbolTable.
```

Comment:
It would probably be good to briefly describe why we don't store strings but rather IDs for size and uniqueness reasons.

Reply:
The fact that we keep the symbol table pointer is something I still disagree with, but I first have to look at the filesystem implementation attachment.
It indicates that the user of the `FileId` does not know which SymbolTable it belongs to.

Why do we even expect it to be in different SymbolTables ? It sounds like we want one SymbolTable in the filesystem implementation, where only the filesystem hands out new IDs. Since that is a singleton, it is also clear where that SymbolTable belongs to. That way, we also don't have the confusing mish-mash of actual symbols and filesystem-strings in the SymbolTable as it is now.

(we should probably rename SymbolTable at some point to string-unifier or SymbolFactory like in UHDM, because it is not really a SymbolTable, but that is for a separate change :) )